superset/connectors/sqla/models.py

```diff
@@ -224,10 +224,10 @@ class TableColumn(Model, BaseColumn, CertificationMixin):
     __tablename__ = "table_columns"
     __table_args__ = (UniqueConstraint("table_id", "column_name"),)
     table_id = Column(Integer, ForeignKey("tables.id"))
-    table: SqlaTable = relationship(
+    table = relationship(
         "SqlaTable",
-        backref=backref("columns", cascade="all, delete-orphan"),
-        foreign_keys=[table_id],
+        back_populates="columns",
+        lazy="joined",  # Eager loading for efficient parent referencing.
```

Comment:
The default is `lazy="select"` which results in lazy loading and [N +1](https://docs.sqlalchemy.org/en/14/glossary.html#term-N-plus-one-problem) `SELECT` statements.